co-authors-plus.php

```diff
@@ -1048,23 +1048,27 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	 * @return int Post count
 	 */
 	function filter_count_user_posts( $count, $user_id ) {
-		$user = get_userdata( $user_id );
+		$user     = get_userdata( $user_id );
+		$coauthor = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
 
-		if ( is_object( $user ) ) {
-			$user = $this->get_coauthor_by( 'user_nicename', $user->user_nicename );
+		// Return $count, if no coauthor exists.
+		if ( ! is_object( $coauthor ) ) {
+			return $count;
+		}
 
-			$term = $this->get_author_term( $user );
+		$term = $this->get_author_term( $coauthor );
```

Comment:
Do you see an opportunity from this line to practice defensive coding below?

Reply:
I was under the impression that `is_int( $term->count )` in https://github.com/Automattic/Co-Authors-Plus/blob/c9e3f660c2629dfb288f2ae8b3109876c2890564/co-authors-plus.php#L1062 and https://github.com/Automattic/Co-Authors-Plus/blob/c9e3f660c2629dfb288f2ae8b3109876c2890564/co-authors-plus.php#L1067, the sections you mentioned before, are defensive coding for `$term`.

Reply:
Ah, I see what you mean. I changed `is_int( $term->count )` to `is_object( $term->count )` to ensure that $term is an object. I kept `strlen( $coauthor->linked_account ) > 2 )` for now, as this check is needed to distinguish if an account is linked.